system/Session/Session.php

```diff
@@ -511,7 +511,22 @@ public function get(string $key = null)
 
 		return $userdata;
 	}
-
+	/**
+	 * Push a value into a session array.
+	 *
+	 * @param  string  $key
+	 * @param  mixed   $value
+	 * @return void
+	 */
+	public function push($key, $value)
+	{
+		if ($this->has($key)) 
+		{
+			$array = $this->get($key);
+			$array = array_merge($array, $value);
```

Comment:
Should have a check to ensure it's an array or throw an exception. I realize that array_merge will throw it's own language, but this gives us a chance to customize the message.

Reply:
ok changes would be made